### ISSUE TITLE
`gum log -l fatal` no longer returns error status code

### ISSUE THREAD

**Issue by alexschcom** [charmbracelet/gum#574]
**Describe the bug**
Calling `gum log -l fatal <...>` used to return an error status code which would stop execution of further commands.

**To Reproduce**
1. Execute `gum log -l fatal hello; echo world`
2. gum logs fatal msg but continues to execute `echo`.

**Expected behavior**
the `log` library calls `os.Exit(1)` on `log.Fatal(err)`. Correspondingly, using `gum log -l fatal` should also stop execution of the script.

**Desktop (please complete the following information):**
 - OS: macOS
 - Version 14.4.1

**Comment by MikaelFangel** [comment]
I checked and `gum log -l fatal hello` indeed returns an error status code. More precisely it returns 1. (See the image)

![Screenshot 2024-05-13 at 13 46 07](https://github.com/charmbracelet/gum/assets/34864484/7756a8c2-22fa-4011-b6e0-dbf2ed9b6bc1)

You should use `&&` instead of `;` between your commands. The reason is that `&&` ensures the second command only runs if the first command succeeds. This can help prevent potential issues if the first command fails for some reason.

**Comment by MikaelFangel** [comment]
I have tried to checkout the tags v0.13.0 and v0.12.0 and I can't reproduce the behaviour you're talking about. Can you please provide some more details like a version number and a reproducible example of the behaviour you think have changed.

Also you can stop the script by checking if the error code is 1.

**Comment by alexschcom** [comment]
I am using version `0.14.1`. I remember before the `0.14.x` update, calling `gum log -l fatal` would stop execution of following commands in a script. For example:
```
gum log -l fatal "Some error occurred"
echo hello
```
Here, the `echo` command would **not** execute, since `gum log -l fatal` would automatically exit the script, as it does in Go files ([see here](https://github.com/charmbracelet/log?tab=readme-ov-file#levels)). This makes sense, considering we're dealing with a **fatal** error. Now, however, it **does** execute the following commands...

**Comment by piero-vic** [comment]
@alexschcom You can add `set -e` at the begining of your script to exit immediately if a command fails.

```bash
#!/usr/bin/env bash

set -e

gum log -l fatal "Some error occurred"
echo hello
```

**Comment by caarlos0** [comment]
seems to be working on main.

**Comment by alexschcom** [comment]
What seems to be working?

<img width="928" alt="Screenshot 2024-07-26 at 3 29 11 PM" src="https://github.com/user-attachments/assets/a27e1a6d-d0d8-4aa6-be9f-fdc94e900ef2">

**Comment by caarlos0** [comment]
on main, not released yet.

**Comment by alexschcom** [comment]
Any updates on this? Still doesn't work correctly.

<img width="928" alt="Screenshot 2024-11-30 at 12 59 46 PM" src="https://github.com/user-attachments/assets/4d9f4f36-ee1e-48ec-92b8-b4099ae8910f">

**Comment by caarlos0** [comment]
@alexschcom your script probably needs a `set -e`... gum is working properly, you can verify that by running `gum log -l fatal` directly on your shell.

**Comment by alexschcom** [comment]
@caarlos0 the `log` library calls `os.Exit(1)` on `log.Fatal(err)`. Correspondingly, using `gum log -l fatal` should also stop further execution of the script, since when a _fatal_ error is encountered, there's by definition no sense in continuing execution. Note that this is how `gum` used to work!

**Comment by caarlos0** [comment]
`gum log -l fatal` still exits 1:

```console
$ gum --version
gum version 0.16.0

$ gum log -l fatal lala && echo 'this wont show up'
FATAL lala
```

**Comment by alexschcom** [comment]
@caarlos0 I'm sorry but you're wrong.

No one uses `&&` after logging a _fatal_ error, so what you said may be technically correct but practically useless. You'd log a potential error and continue with the next thing:

```
gum log -l fatal "error"
echo "hello"
```

still outputs `hello`. Which doesn't make sense, since we've encountered a _fatal_ error. Again, `gum` _used to_ `return` an error in this case, stopping execution of the rest of the script.

I'm aware setting `-e` mimics the expected behavior, but semantically speaking, it's a workaround.

**Comment by caarlos0** [comment]
@alexschcom no I'm not.

What you asking for never worked, and can't never work. It's how bash/zsh/etc work.
The only way that could work would be to kill the parent PID, which we never did, and is probably a bad idea.

> Again, gum used to return an error in this case, stopping execution of the rest of the script.

The concept of errors in bash/zsh/etc does not exist. It's just `exit` codes. 
You can't "return" an error from a script/binary, you can only `exit <code>`, and Gum does exactly that: it `exit 1` on `log -l fatal` - as you can see from my previous comments. 

The same behavior happens if you create a script that `exit 1`:

```console
$ echo 'exit 1' >foo

$ chmod +x foo

$ ./foo; echo hi
hi

$ ./foo && echo hi
```

This is also how every coreutils binary works:

```console
$ rm file-that-doest-exist; echo hi
rm: cannot remove 'file-that-doest-exist': No such file or directory
hi
```

I also downloaded gum 0.13, also with the same behavior:

```console
$ ./gum log -l fatal "something bad"; echo hello
FATAL something bad
hello
```

You say this used to work, I don't see how. My guess is that you had some bash option set globally.

---

TL;DR: either add a `set -e` to the beginning of your script, or `exit 1` after calling `gum log -l fatal`, or run `gum log -l fatal "text" && next command`. 
Assuming it's a script, I would recommend adding the `set -e`. In fact, I would recommend to add `set -euo pipefail` to pretty much every bash script. [link](https://duckduckgo.com/?q=bash+best+practices+%2B+%22set+-e%22)
